### PR TITLE
go 1.15.7

### PR DESCRIPTION
**Pull request by leonklingele**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

### CODE REVIEW

File: Formula/go.rb [hunk 1]
```diff
@@ -7,11 +7,11 @@ class Go < Formula
     if Hardware::CPU.arm?
       url "https://golang.org/dl/go1.16beta1.src.tar.gz"
       sha256 "48e032c8cf71af4dc8119a29ee829c4fbd5265e32fd012564d4a70bb207695c1"
-      version "1.15.6"
```
Comment (leonklingele) [inline Formula/go.rb:10]: This looks quite sketchy, what about `1.16beta1` instead?

Comment (carlocab) [reply Formula/go.rb:10]: That won't work, and will prevent any bottles from being provided on ARM.

You actually need to change this to 1.15.7.

Comment (leonklingele) [reply Formula/go.rb:10]: That's exactly what I did already :)

**Comment by carlocab** [comment]
Big Sur:
```
Error: 3 failed steps!
brew install --build-from-source charm-tools
brew install --build-from-source grv
brew install --build-from-source kustomize
```

Catalina:
```
Error: 2 failed steps!
brew install --build-from-source charm-tools
brew install --build-from-source kustomize
```

Mojave:
```
Error: 3 failed steps!
brew install --build-from-source charm-tools
brew install --build-from-source grv
brew install --build-from-source kustomize
```

**Comment by carlocab** [comment]
ARM:

<details><summary>Error: 192 failed steps!</summary>
<p>

```
Error: 192 failed steps!
brew install --build-from-source ahoy
brew install --build-from-source --only-dependencies akamai
brew install --build-from-source akamai
brew install --build-from-source amazon-ecs-cli
brew install --build-from-source apache-brooklyn-cli
brew install --build-from-source aptly
brew install --build-from-source arduino-cli
brew install --build-from-source argo
brew install --build-from-source assh
brew install --build-from-source auditbeat
brew test --retry --verbose awsweeper
brew test --retry --verbose bazelisk
brew install --build-from-source --only-dependencies bettercap
brew install --build-from-source bettercap
brew install --build-from-source bitrise
brew install --build-from-source buildifier
brew install --build-from-source buildozer
brew install --build-from-source caddy
brew install --build-from-source carina
brew install --build-from-source cartridge-cli
brew install --build-from-source ccat
brew install --build-from-source cf-tool
brew install --build-from-source charm
brew install --build-from-source --only-dependencies charm-tools
brew install --build-from-source charm-tools
brew test --retry --verbose chart-testing
brew install --build-from-source circleci
brew install --build-from-source --only-dependencies clair
brew install --build-from-source clair
brew test --retry --verbose cloudformation-cli
brew install --build-from-source --only-dependencies collector-sidecar
brew install --build-from-source collector-sidecar
brew install --build-from-source confd
brew test --retry --verbose conjure-up
brew install --build-from-source consul
brew install --build-from-source cosi
brew install --build-from-source cql
brew install --build-from-source --only-dependencies cwlogs
brew install --build-from-source cwlogs
brew install --build-from-source devd
brew install --build-from-source --only-dependencies devspace
brew install --build-from-source devspace
brew install --build-from-source docker-credential-helper
brew install --build-from-source docker-credential-helper-ecr
brew install --build-from-source docker-gen
brew install --build-from-source --only-dependencies docker-machine-driver-hyperkit
brew install --build-from-source docker-machine-driver-hyperkit
brew install --build-from-source docker-machine-driver-vmware
brew install --build-from-source docker-machine-driver-vultr
brew install --build-from-source docker-swarm
brew install --build-from-source docker2aci
brew install --build-from-source dockerize
brew install --build-from-source --only-dependencies dockviz
brew install --build-from-source dockviz
brew install --build-from-source dockward
brew test --retry --verbose driftctl
brew install --build-from-source drone-cli
brew install --build-from-source dvm
brew install --build-from-source emp
brew install --only-dependencies --include-test envconsul
brew test --retry --verbose envconsul
brew install --build-from-source envoy
brew test --retry --verbose etcd
brew install --build-from-source --only-dependencies fabio
brew install --build-from-source fabio
brew install --build-from-source filebeat
brew install --build-from-source flint-checker
brew install --build-from-source flux
brew install --build-from-source --only-dependencies frugal
brew install --build-from-source frugal
brew install --build-from-source fsql
brew install --build-from-source gdm
brew install --build-from-source gdrive
brew install --build-from-source --only-dependencies geoipupdate
brew install --build-from-source geoipupdate
brew install --build-from-source --only-dependencies git-annex-remote-rclone
brew install --build-from-source git-annex-remote-rclone
brew install --build-from-source git-appraise
brew install --build-from-source git-sizer
brew install --build-from-source github-release
brew install --build-from-source gollum
brew install --build-from-source gotags
brew install --build-from-source gotop
brew test --retry --verbose gox
brew test --retry --verbose gpm
brew install --build-from-source gron
brew install --build-from-source grv
brew install --build-from-source gx-go
brew install --build-from-source heartbeat
brew install --build-from-source --only-dependencies helmsman
brew install --build-from-source helmsman
brew install --build-from-source hivemind
brew install --build-from-source httpdiff
brew install --build-from-source ioctl
brew install --build-from-source --only-dependencies ironcli
brew install --build-from-source ironcli
brew install --build-from-source istioctl
brew install --build-from-source --only-dependencies jabba
brew install --build-from-source jabba
brew install --build-from-source jp
brew install --build-from-source jsonpp
brew install --build-from-source kedge
brew test --retry --verbose keptn
brew install --build-from-source --only-dependencies kops
brew install --build-from-source kops
brew install --build-from-source --only-dependencies krew
brew install --build-from-source krew
brew install --build-from-source ksync
brew install --build-from-source kube-linter
brew install --build-from-source --only-dependencies kube-ps1
brew install --build-from-source kube-ps1
brew install --build-from-source --only-dependencies kubectx
brew install --build-from-source kubectx
brew install --build-from-source --only-dependencies kubeless
brew install --build-from-source kubeless
brew install --build-from-source kubernetes-cli
brew install --only-dependencies --include-test kubie
brew test --retry --verbose kubie
brew install --build-from-source kumactl
brew install --build-from-source kustomize
brew install --build-from-source launch_socket_server
brew install --build-from-source --only-dependencies leaps
brew install --build-from-source leaps
brew install --build-from-source lunchy-go
brew install --build-from-source mailhog
brew install --build-from-source massren
brew install --build-from-source metricbeat
brew install --build-from-source --only-dependencies microplane
brew install --build-from-source microplane
brew install --build-from-source --only-dependencies minikube
brew install --build-from-source minikube
brew install --build-from-source modd
brew install --build-from-source mpdviz
brew install --build-from-source octant
brew install --build-from-source packer
brew install --build-from-source packetbeat
brew test --retry --verbose packr
brew install --build-from-source path-extractor
brew install --build-from-source pgweb
brew install --build-from-source --only-dependencies piknik
brew install --build-from-source piknik
brew install --build-from-source pilosa
brew install --build-from-source prometheus
brew install --build-from-source pup
brew install --build-from-source qpm
brew install --build-from-source rack
brew install --build-from-source rancher-compose
brew install --build-from-source --only-dependencies riff
brew install --build-from-source riff
brew install --build-from-source s-search
brew install --build-from-source scc
brew test --retry --verbose server-go
brew install --build-from-source --only-dependencies shellz
brew install --build-from-source shellz
brew install --build-from-source sift
brew install --build-from-source snag
brew install --build-from-source --only-dependencies snap-telemetry
brew install --build-from-source snap-telemetry
brew install --build-from-source source-to-image
brew install --build-from-source spaceinvaders-go
brew install --build-from-source --only-dependencies ssh-vault
brew install --build-from-source ssh-vault
brew install --build-from-source --only-dependencies step
brew install --build-from-source step
brew install --only-dependencies --include-test stolon
brew test --retry --verbose stolon
brew install --build-from-source stout
brew install --build-from-source sync_gateway
brew test --retry --verbose syncthing
brew install --build-from-source --only-dependencies tanka
brew install --build-from-source tanka
brew install --build-from-source td
brew install --build-from-source teleconsole
brew install --build-from-source termshare
brew test --retry --verbose termshark
brew test --retry --verbose terraform
brew install --build-from-source terraform-docs
brew install --build-from-source terraform-inventory
brew install --build-from-source --only-dependencies terraform-provider-libvirt
brew install --build-from-source terraform-provider-libvirt
brew install --build-from-source --only-dependencies textql
brew install --build-from-source textql
brew install --build-from-source traefik@1
brew install --build-from-source uru
brew test --retry --verbose v2ray
brew install --build-from-source --only-dependencies vault
brew install --build-from-source vault
brew install --build-from-source --only-dependencies vert
brew install --build-from-source vert
brew install --build-from-source virgil
brew install --build-from-source vultr
brew install --build-from-source wego
```

</p>
</details>

**Comment by carlocab** [comment]
The `kustomize` failure looks unrelated:
```
==> brew install --build-from-source kustomize
==> FAILED
==> Cloning https://github.com/kubernetes-sigs/kustomize.git
Cloning into '/Users/brew/Library/Caches/Homebrew/kustomize--git'...
Updating files:  16% (2110/12929)
[snip]
Updating files: 100% (12929/12929), done.
==> Checking out tag kustomize/v3.9.2
HEAD is now at e98eada Merge pull request #3476 from kubernetes-sigs/pinToApi
Error: An exception occurred within a child process:
  RuntimeError: Not a Git repository: /private/tmp/kustomize-20210120-57844-1m0guo0/kustomize
```
This happens in all four CI nodes. Seems like this is more of a problem with the Git API? Ping @SeekingMeaning

**Comment by SeekingMeaning** [comment]
Ack. Yes, kustomize should be calling Utils.git_head in the root buildpath directory

**Comment by carlocab** [comment]
`charm-tools` fails for me locally as well, without this PR. Same error as in the CI nodes. It doesn't even need `go` to build, it only depends on `go` through a dependency on `charm`.

**Comment by carlocab** [comment]
Testing `charm-tools` and `grv` on GitHub runners: https://github.com/carlocab/homebrew-personal/pull/29

**Comment by carlocab** [comment]
`kustomize` should've been fixed in #69464.

`charm-tools` fails on a GitHub Catalina runner: https://github.com/carlocab/homebrew-personal/pull/29/checks?check_run_id=1740409343

Big Sur runner still queued.

**Comment by carlocab** [comment]
`charm-tools` and `grv` also don't build with the existing formula on Big Sur: https://github.com/carlocab/homebrew-personal/runs/1743071106?check_suite_focus=true

Errors on the Mojave CI node also look similar to the errors I'm seeing on Catalina/Big Sur (both in the test runs here and on the GitHub runners with the old Go formula).

**Review by carlocab — APPROVED** [review]
Regressions don't look like they were caused by this PR.

Thanks, @leonklingele!

**Comment by andig** [comment]
Ahh, I'm 17minutes to late to open this PR- thank you!

**Comment by carlocab** [comment]
@leonklingele opened this PR two days ago, so 17 minutes earlier wouldn't have helped. 🙃

We could use some with trying to fix `charm-tools` and `grv`, if you have the time.

**Comment by andig** [comment]
Grv is deprecated license. Is a valid fix to replace gpl3 with just gpl?

**Comment by andig** [comment]
The build issue for GRV is here: https://github.com/rgburke/grv/issues/107

**Comment by carlocab** [comment]
>Grv is deprecated license. Is a valid fix to replace gpl3 with just gpl?

If it's still GPL3 then it needs to be replaced with `GPL-3.0-only` or `GPL-3.0-or-later`, depending on which one is accurate. If neither are accurate we need to replace it with the correct license identifier.

> The build issue for GRV is here: rgburke/grv#107

Aha; thanks for tracking that down. It seems to be a recurring build problem. We may need to disable it at some point.

**Comment by carlocab** [comment]
Thanks for your investigation, @andig. I've disabled `grv` at https://github.com/Homebrew/homebrew-core/pull/69633.

Now we only need to sort out `charm-tools`.

**Comment by andig** [comment]
Charm looks like https://github.com/Homebrew/homebrew-core/pull/68532?

**Pull request closed without merging**